pkg/controllers/sidecar/config/config.go

```diff
@@ -244,18 +246,28 @@ func (s *ScyllaConfig) setupEntrypoint(ctx context.Context) (*exec.Cmd, error) {
 		args["cpuset"] = &cpusAllowed
 	}
 
+	version, err := semver.Parse(cluster.Spec.Version)
+	if err != nil {
+		version = v1.ScyllaDefaultVersion
+		s.logger.Error(ctx, "The operator couldn't recognize the current scylla version, using default version",
+			"spec_version", cluster.Spec.Version, "default_version", version)
+	}
+	s.logger.Info(ctx, "Scylla version detected", "version", version)
+
 	if len(cluster.Spec.ScyllaArgs) > 0 {
-		version, err := semver.Parse(cluster.Spec.Version)
-		if err != nil {
-			s.logger.Info(ctx, "This scylla version might not support ScyllaArgs", "version", cluster.Spec.Version)
-			appendScyllaArguments(ctx, s, cluster.Spec.ScyllaArgs, args)
-		} else if version.LT(v1.ScyllaVersionThatSupportsArgs) {
-			s.logger.Info(ctx, "This scylla version does not support ScyllaArgs. ScyllaArgs is ignored", "version", cluster.Spec.Version)
+		if version.LT(v1.ScyllaVersionThatSupportsArgs) {
+			s.logger.Info(ctx, "This scylla version does not support ScyllaArgs. ScyllaArgs is ignored", "version", version)
 		} else {
 			appendScyllaArguments(ctx, s, cluster.Spec.ScyllaArgs, args)
 		}
 	}
 
+	if _, err := os.Stat(scyllaIOPropertiesPath); err == nil && version.GTE(v1.ScyllaVersionThatSupportsDisablingIOTuning) {
```

Comment:
```
func scyllaSupportsDisabledIOSetup(v string) bool {
    version, err := semver.Parse(v)
    if err != nil {
        return false
    }
    _, err := os.Stat(scyllaIOPropertiesPath)
    return err == nil && version.GTE(v1.ScyllaVersionThatSupportsDisablingIOTuning) 
}
```
This way we don't need this error prone default version assumption.

Reply:
I added a wrapper on top of the version instead